### PR TITLE
fix the reconcilation script to rebuild cases as well

### PR DESCRIPTION
**Pull request by Rohit25negi**
condition `if doc_id not in docs_in_ucr:` was only allowing the missing forms to rebuild not the outdated cases. Because cases can be existing in the UCR and yet get outdated. 
This is identified while debugging this ticket: https://dimagi-dev.atlassian.net/browse/ICDS-1290.

I found that there were some child cases that are open in the UCR and their corresponding person case in the UCR is closed. But in the HQ both child health case and its person case is closed.
I tried republishing the close form of one child case and it successfully reached to the ucr. So I think its just a general case of cases-updates not reaching to UCR same as forms and not specific to person and child case. 
 
After this is merged, This reconciliation script will need to rerun only for cases

### CODE REVIEW

File: custom/icds_reports/tasks.py [hunk 1]
```diff
@@ -1745,20 +1745,29 @@ def get_data_not_in_ucr(status_record):
...
         docs_in_ucr = _get_docs_in_ucr(domain, status_record.table_id, doc_ids)
         for doc_id, doc_subtype, sql_modified_on in chunk:
-            if doc_id not in docs_in_ucr:
+            in_ucr = False
+            for ucr_doc in docs_in_ucr:
+                if ucr_doc[0] == doc_id:
+                    in_ucr = True
+                    # For the cases which are outdated.  This condition also handles the time drift of 1 sec
+                    # between main db and ucr db. i.e  doc will even be included when inserted_at-sql_modified_on <= 1 sec
+                    if sql_modified_on - ucr_doc[1] >= timedelta(seconds=-1):
```
Comment (Rohit25negi) [inline custom/icds_reports/tasks.py]: Added this `if` condition based on my best understanding of the time drifts between the database. Refering (https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/management/commands/stale_data_in_ucr.py#L102) which may be wrongly handling it

**Review by calellowitz — CHANGES_REQUESTED** [review]
I think there are some possible performance problems with this code that can be addressed while also simplifying the logic a bit

File: custom/icds_reports/tasks.py [hunk 2]
```diff
@@ -1745,20 +1745,29 @@ def get_data_not_in_ucr(status_record):
         doc_ids = [val[0] for val in chunk]
         docs_in_ucr = _get_docs_in_ucr(domain, status_record.table_id, doc_ids)
         for doc_id, doc_subtype, sql_modified_on in chunk:
-            if doc_id not in docs_in_ucr:
+            in_ucr = False
```
[the remark this answers is not in the document]
Comment (calellowitz) [reply custom/icds_reports/tasks.py]: Might be good to also include your comment in my suggested change because I agree its a bit confusing

**Comment by Rohit25negi** [comment]
@calellowitz  Simplified the code as suggested, please check

File: custom/icds_reports/tasks.py [hunk 3]
```diff
@@ -1743,22 +1743,27 @@ def get_data_not_in_ucr(status_record):
     chunk_size = 1000
     for chunk in chunked(matching_records_for_db, chunk_size):
         doc_ids = [val[0] for val in chunk]
-        docs_in_ucr = _get_docs_in_ucr(domain, status_record.table_id, doc_ids)
+        doc_id_and_inserted_in_ucr = _get_docs_in_ucr(domain, status_record.table_id, doc_ids)
```
Comment (calellowitz) [inline custom/icds_reports/tasks.py]: merging this now so we can get it out but can you change this name back. I think the new name is much more confusing that the old.

**Review by calellowitz — APPROVED** [review]

**Pull request merged**